onnxruntime/core/session/inference_session.cc

```diff
@@ -1934,6 +2000,15 @@ Status InferenceSession::Run(const RunOptions& run_options,
 #ifdef ONNXRUNTIME_ENABLE_INSTRUMENT
   TraceLoggingWriteStop(ortrun_activity, "OrtRun");
 #endif
+
+  // As two inference runs (one for memory allocation and one for graph capturing)
+  // are needed before replaying the captured graph, here run the inference again
+  // to capture the graph, so that users just need one session run to capture
+  // the graph.
+  if (cached_execution_provider_for_graph_replay_.IsGraphCaptureEnabled() &&
```

Comment:
Should we only do the recursive Run() call only if the first Run() was successful (i.e.) retval didn't have a failure status ? 
Also, can we add an INFO log to capture that we are starting another Run() and the reason we are doing so ?

Reply:
Done